lib/logflare_web/controllers/source_controller.ex

```diff
@@ -556,16 +556,9 @@ defmodule LogflareWeb.SourceController do
     log_events = RLS.list_for_cluster(source.token)
 
     for le <- log_events, le do
-      le =
-        le
-        |> Map.from_struct()
-        |> Map.take([:body, :via_rule, :origin_source_id])
-
-      if le.via_rule do
-        %{le | via_rule: %{regex: le.via_rule.regex}}
-      else
-        le
-      end
+      le
+      |> Map.from_struct()
+      |> Map.take([:body, :via_rule, :origin_source_id])
```

Comment:
```suggestion
      Map.take(le, [:body, :via_rule, :origin_source_id])
```

Struct is map already, no need to do that double editing.